app/jobs/account_registration_job.rb

```diff
@@ -0,0 +1,12 @@
+class AccountRegistrationJob < ApplicationJob
+  queue_as :default
+
+  def perform(create_params)
```

Comment:
This is done in an async job to prevent account enumeration via timing attacks.

Reply:
Why would an async job prevent this?

Reply:
When you register, we need to check if a user with that email address already exists to either:
- create a new user entry and send the confirmation email; or
- send an email saying that an account already exists, and asking the user to either login or recover the password

This is done (instead of saying "hey you already have an account" in the webpage) to prevent anyone from figuring out who is registered or not in the website by registering a new account.

Even with not displaying the error message, an attacker could measure the response time to try to figure out which branch the server had taken. This wouldn't be super precise, since you only have one shot, but I decided to async it to prevent it altogether. Email sending would have to be done async anyway to avoid stalling the main http loop with the email API requests, so I ended up moving everything into the async task (We're sending all the other emails using an async task by using deliver_later).

More info on timing attacks, slightly related: https://soatok.blog/2021/08/20/lobste-rs-password-reset-vulnerability/